src/pdn/src/straps.cpp

```diff
@@ -383,6 +385,59 @@ void FollowPins::makeShapes(const Shape::ShapeTreeMap& other_shapes)
   const int x_start = boundary.xMin();
   const int x_end = boundary.xMax();
   odb::dbTechLayer* layer = getLayer();
+
+  std::vector<std::shared_ptr<Shape>> power_obs, ground_obs;
+  if (getExtendMode() == CLIP_CELLS) {
+    for (odb::dbInst *inst : grid->getBlock()->getInsts()) {
+      if (!inst->isPlaced() || !boundary.contains(inst->getBBox()->getBox())) {
+        continue;
+      }
+
+      const odb::dbTransform transform = inst->getTransform();
+
+      for (odb::dbITerm *iterm : inst->getITerms()) {
+
+        bool is_power = iterm->getNet() == power;
+        bool is_ground = iterm->getNet() == ground;
+
+        // Non-power pins might obstruct
```

Comment:
This will generate a huge amount of obstructions.  Have you looked at the runtime/memory impact?

Reply:
As this is being run straight after tapcell, there aren't that many placed cells other than those - I wasn't noticing any significant impact on our medium (~100k cells) design.

Reply:
That's true in ORFS but nothing prevents someone from loading a design with pre-placed cells or many macros.  Would it be reasonable to limit the check to endcaps?